src/Halogen/HTML/Events/Forms.purs

```diff
@@ -1,48 +1,34 @@
--- | Convenience functions for working with form elements.
-
+-- | Convenience functions for working with form events.
 module Halogen.HTML.Events.Forms
-  ( onValueChanged
+  ( onValueChange
+  , onValueInput
   , onChecked
-  , onInput
   ) where
 
 import Prelude
 
-import DOM
-
-import Data.Maybe
-import Data.Either
-import Data.Foreign
-import Data.Foreign.Class
-import Data.Traversable (traverse)
+import Data.Either (either)
+import Data.Foreign (toForeign)
+import Data.Foreign.Class (IsForeign, readProp)
 
-import Control.Plus
-import Control.Alternative
+import Halogen.HTML.Core (Prop(), eventName, handler)
+import Halogen.HTML.Events.Handler (EventHandler())
 
-import Halogen.HTML.Events.Handler
+-- | Attach event handler to event `key` with getting `prop` field as an
+-- | argument of `handler`.
+addForeignPropHandler :: forall f value. (Applicative f, IsForeign value) => String -> String -> (value -> EventHandler (f Unit)) -> Prop (f Unit)
```

Comment:
I was working under the assumption that `target.value` gave values of different types based on the input type: checkbox, radio, text, etc., but that's not true. According to the spec, it's always a string: http://www.w3schools.com/jsref/prop_attr_value.asp

So if the implementation is sound and the `value` property isn't set artificially, by a plugin for example, there's no need for a polymorphic `value`. Validating the value is probably still a good idea though, in case of bad plugins or whatever.